content/blog/prototyping/index.md

```diff
@@ -0,0 +1,767 @@
++++
+title = "Prototyping in Rust"
+date = 2024-12-03
+draft = false
+template = "article.html"
+[extra]
+# hero = "hero.svg"
+series = "Idiomatic Rust"
+resources = [
+    "[Throw away code - Vorner's blog](https://vorner.github.io/2020/09/20/throw-away-code.html)",
+    "[RustConf 2021 - The Importance of Not Over-Optimizing in Rust by Lily Mara](https://www.youtube.com/watch?v=CV5CjUlcqsw)",
+    "[Rust Nation UK - Easy Mode Rust by Andre Bogus](https://www.youtube.com/watch?v=33FG6O3qejM)",
+]
+reviews = [
+    { name = "Theodor-Alexandru Irimia", url = "https://github.com/tirimia" },
+]
++++
+
+Programming is an iterative process - as much as we like to come up with the perfect solution from the start, it rarely works that way.
+
+Whether you're writing games, CLI tools, or designing library APIs, prototyping helps tremendously in finding the best approach before committing to a design.
+It helps reveal the patterns behind more idiomatic code.
+
+Contrary to popular belief, Rust is a joy for building prototypes.
+
+For all its explicitness, Rust is surprisingly ergonomic when iterating on ideas.
+
+You don't need to be a Rust expert to be productive - in fact, many of the techniques we'll discuss specifically help you *sidestep* Rust's more advanced features.
+If you focus on simple patterns and make use of Rust's excellent tooling, even less experienced Rust developers can quickly bring their ideas to life.
+
+## Why People Think Rust Is Not Good For Prototyping
+
+The common narrative goes like this:
+
+> When you start writing a program, you don't know what you want and you change your mind pretty often.
+> Rust pushes back when you change your mind because the type system is very strict.
+> On top of that, getting your idea to compile takes longer than in other languages, so the feedback loop is slower.
+
+That's why people believe that Rust isn't a good fit for prototyping.
+
+I've found that developers from other languages and Rust beginners often share this preconception about Rust.
+These developers stumble over the strict type system and the borrow checker while trying to sketch out a solution.
+They believe that with Rust you're either at 0% or 100% done (everything works and has no undefined behavior) and there's nothing in between.
+
+Here are some typical misbeliefs: 
+
+1. "Memory safety and prototyping just don't go together."
+2. "Ownership and borrowing take out the fun of prototyping."
+3. "You have to get all the details right from the beginning."
+4. "Rust always requires you to handle errors."
+
+These are all [common misconceptions](https://medium.com/@victor.ronin/love-hate-relationship-with-rust-language-part-2-c36f57d5485d) and they are not true. 
+
+**It turns out you can avoid all of these pitfalls and still get a lot of value from prototyping in Rust.**
+
+## What Makes Rust Great for Prototyping?
```

Comment:
I'm not so sure about this section and its placement. You already introduced what you present as misconceptions, and now you're just telling me the opposite is true, without further proof.

I'm not sure what to do instead. Maybe shorten this (there's a redundant list further down anyway, as I'm calling out there), or explicitly call out that the purpose of this section is tell the reader why they should bother with Rust for prototyping, and that lots of proof will follow.

Or maybe consider this section (maybe even an expanded form of it) for an appendix, or a follow-up article?